Assets/MixedRealityToolkit-SDK/Features/Input/Scripts/InputSystemGlobalListener.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Internal.Managers;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.Input
+{
+    /// <summary>
+    /// This component ensures that all input events are forwarded to this <see cref="GameObject"/> when focus or gaze is not required.
+    /// </summary>
+    public class InputSystemGlobalListener : MonoBehaviour
+    {
+        private IMixedRealityInputSystem inputSystem = null;
+        private IMixedRealityInputSystem InputSystem => inputSystem ?? (inputSystem = MixedRealityManager.Instance.GetManager<IMixedRealityInputSystem>());
```

Comment:
Maybe we should put this code in a generic base class?  As we seem to be copying it everywhere.

Reply:
Maybe, it's just the easiest way to get the input system reference.

Reply:
I'm not sure making a base input class would be good for the SDK components. I just wanted then to be simple and self contained.

Reply:
Nothings more self contained that a MonoBehavjour base class which implements the same code for all dependant classes.
Also make it a heck of a lot easier for people to create their own classes as there is less to copy.  They just use the new base class instead of MonoBehaviour

Reply:
BaseInputHandler it is then.